docs/templates.md

```diff
@@ -2,26 +2,36 @@
 
 ## Template
 ### Templates using decorators
+A **template** in Pyccel, is used to allow the same function to take arguments of different types from a selection of types the user specifies.
 #### The usage
+In this example the argument **a**, could either be an integer or float, and the same for the argument **b**:
 ```python
 from pyccel.decorators import template
-@template(name='T', types=['int','real'])
+@template(name='T', types=['int','float'])
+@template(name='Z', types=['int','float'])
 def f(a : 'T', b : 'T'):
 	pass
 ```
```

Comment:
If I understand this example correctly, the argument `b` should be of type `'Z'`.